File: LeetCode Solutions/April 2022 LeetCoding Challenge/Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maximumArea, i, j = 0, 0, len(height) - 1
        while(i < j):
            maximumArea = max(maximumArea, min(height[i], height[j]) * (j - i))
            if(height[i] < height[j]):
                i += 1
            else:
                j -= 1
        return maximumArea

File: LeetCode Solutions/LeetCoding Challenge/February 2021 LeetCoding Challenge/Container With Most Water/Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxArea = 0, i = 0, j = height.size() - 1;
        while (i < j){
            maxArea = max(maxArea, min(height[i], height[j]) * (j-i));
            if (height[i] < height[j]){
                i += 1;
            }
            else{
                j -= 1;
            }
        }
        return maxArea;
    }
};